pandas/core/internals/concat.py

```diff
@@ -245,6 +247,38 @@ def concatenate_managers(
     return BlockManager(tuple(blocks), axes)
 
 
+def _reindex_columns_void(
```

Comment:
i would rename to `_maybe_reindex_columns_na_proxy`, e.g. don't use the term `void` as we don't use it elsewhere and instead use `na_proxy` (I know the dtype is V which is fine); alt make this clear in the doc-string here.

Reply:
changed + green